libnmstate/nm/dns.py

```diff
@@ -179,3 +191,18 @@ def _get_ip_profile_dns_config(ip_profile):
         DNS.SEARCH: ip_profile.props.dns_search,
         DNS_METADATA_PRIORITY: ip_profile.props.dns_priority,
     }
+
+
+def _get_auto_dns_false_iface(family, iface_states):
+    for iface_name, iface_state in iface_states.items():
+        ip_state = iface_state.get(family, {})
+        if ip_state.get(InterfaceIP.ENABLED) and not ip_state.get(
+            InterfaceIP.AUTO_DNS
+        ):
+            if family == Interface.IPV4 and ip_state.get(InterfaceIPv4.DHCP):
+                return iface_name
+            if family == Interface.IPV6 and (
+                ip_state.get(InterfaceIPv6.DHCP)
+                or ip_state.get(InterfaceIPv6.AUTOCONF)
+            ):
```

Comment:
You can use `InterfaceIP.DHCP` and make it common and then drop the family type:
In that case, you can just do something like:
```
if ip_state.get(InterfaceIP.DHCP) or ip_state.get(InterfaceIPv6.AUTOCONF):
```

Reply:
Thanks.